website/docs/segments/xmake.mdx

```diff
@@ -0,0 +1,54 @@
+---
+id: xmake
+title: XMake
+sidebar_label: XMake
+---
+
+## What
+
+Display the currently active xmake version.
+
+## Sample Configuration
+
+```json
+{
+  "type": "xmake",
+  "style": "powerline",
+  "powerline_symbol": "\uE0B0",
+  "foreground": "#e0f2f1",
+  "background": "#22a079",
+  "template": " xmake v{{ .Full }} "
+},
+```
+
+## Properties
+
+| Name                   | Type      | Description                                                                                                                                                                          |
+| ---------------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `home_enabled`         | `boolean` | display the segment in the HOME folder or not - defaults to `false`                                                                                                                  |
+| `fetch_version`        | `boolean` | display the xmake version (`xmake --version`) - defaults to `true`                                                                                                                    |
+| `missing_command_text` | `string`  | text to display when the command is missing - defaults to empty                                                                                                                      |
+| `display_mode`         | `string`  | <ul><li>`always`: the segment is always displayed</li><li>`files`: the segment is only displayed when `xmake.lua` file is present (**default**)</li></ul> |
+
+## Template ([into][templates])
+
+:::note default template
+
+```template
+{{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }}
+```
+
+:::
+
+### Properties
+| Name          | Type     | Description                                        |
+| ------------- | -------- | -------------------------------------------------- |
+| `.Full`       | `string` | the full version                                   |
+| `.Major`      | `string` | major number                                       |
+| `.Minor`      | `string` | minor number                                       |
+| `.Patch`      | `string` | patch number                                       |
+| `.Branch`     | `string` | branch name                                        |
```

Comment:
This and the others below won't work out of the box without modifications to the [version](https://github.com/JanDeDobbeleer/oh-my-posh/blob/07acb9b0f6ab8a27de4a970a7d01a58112447305/src/segments/language.go#L24) struct and [logic](https://github.com/JanDeDobbeleer/oh-my-posh/blob/07acb9b0f6ab8a27de4a970a7d01a58112447305/src/segments/language.go#L43).